File: src/main/java/rars/venus/settings/editor/views/PresetsView.java
package rars.venus.settings.editor.views;

import org.jetbrains.annotations.NotNull;
import rars.settings.SettingsTheme;

import javax.swing.*;
import java.util.ArrayList;

public final class PresetsView extends JScrollPane {
    public final @NotNull ArrayList<@NotNull PresetSection> sections;
    private final @NotNull JPanel mainPanel;

    public PresetsView() {
        super();
        this.sections = new ArrayList<>();
        this.mainPanel = new JPanel();
        this.mainPanel.setLayout(new BoxLayout(this.mainPanel, BoxLayout.Y_AXIS));
        this.setViewportView(this.mainPanel);
    }

    public void addSection(final @NotNull PresetSection section) {
        this.sections.add(section);
        this.mainPanel.add(section);
    }

    public static class PresetSection extends JPanel {
        public final @NotNull JButton button;

        public PresetSection(
            final @NotNull String themeName,
            final @NotNull SettingsTheme theme
        ) {
            super();
            this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
            this.button = new JButton("Apply");
            this.add(new JLabel(themeName));
            this.add(Box.createHorizontalGlue());
            this.add(this.button);
        }
    }
}

File: src/main/java/rars/venus/settings/editor/views/PresetsView.kt
package rars.venus.settings.editor.views

import rars.venus.util.BoxLayout
import rars.venus.util.JPanel
import javax.swing.*

class PresetsView : JScrollPane() {
    val sections = mutableListOf<PresetSection>()
    private val mainPanel = JPanel {
        layout = BoxLayout(this, BoxLayout.Y_AXIS)
    }

    init {
        setViewportView(mainPanel)
    }

    fun addSection(section: PresetSection) {
        sections.add(section)
        mainPanel.add(section)
    }

    class PresetSection(themeName: String) : JPanel() {
        val button: JButton = JButton("Apply")

        init {
            BoxLayout(BoxLayout.X_AXIS) {
                +JLabel(themeName)
                horizontalGlue()
                +button
            }
        }
    }
}
